components/canvas/canvas_paint_thread.rs

```diff
@@ -795,6 +790,11 @@ fn is_zero_size_gradient(pattern: &Pattern) -> bool {
     false
 }
 
+fn premultiply_channel_by_alpha(channel: u8, alpha: u16) -> u8 {
+    // add 128 before dividing for more accurate rounding
+    (((channel as u16 * alpha) + 128) / 255) as u8
```

Comment:
Oh, you got a good point here.

The thing is, division is really expensive. Dividing an unsigned number by a power of 2 is not actually treated as dividing, but as a right shift.

Thinking about this math made me want to double check everything, so I wrote a small program to check every possible value - http://is.gd/qYkNvN and it turns out adding 128 is also wrong. Need to add 127, and then divide by 255. Might poke around a little and see if we can replace the division with a shift, but this looks like the most accurate way so far.